Quadriga/src/main/java/edu/asu/spring/quadriga/service/user/mapper/impl/UserDeepMapper.java

```diff
@@ -65,15 +65,49 @@ public IUser getUser(String userName)
             user.setQuadrigaRoles(rolesList);
         } else {
             user = userFactory.createUserObject();
-            quadrigaRole = roleManager
+            IQuadrigaRole quadrigaRole  = roleManager
                     .getQuadrigaRoleByDbId(IQuadrigaRoleManager.MAIN_ROLES, RoleNames.DB_ROLE_QUADRIGA_NOACCOUNT);
+            List<IQuadrigaRole> rolesList = new ArrayList<IQuadrigaRole>();
             rolesList.add(quadrigaRole);
-            user.setQuadrigaRoles(rolesList);
+            user.setQuadrigaRoles(rolesList); 
+        }
+
+        return user;
+    }
+
+    
+    
+    @Transactional
```

Comment:
since the whole class is annotated with transactional, you don't need this here.

Reply:
I don't see @Transactional annotation on the class..The methods are annotated with @Transactional

Reply:
ups, sorry. must have confused that with the previous method.